### DISCUSSION TITLE
Snaps versioning and MetaMask Flask releases

### DISCUSSION THREAD

**Discussion by Vid201** [MetaMask/snaps#1121] [General]
Hey everybody,

We are working with Snaps, and it would be very helpful for the development if there were somewhere defined which Snap version will be included in the next Flask release (any maybe approximate release date). That way, Snap developers can develop for the specific Snaps version ahead of a new Flask, which would make sense since there are breaking changes between different Snaps versions.

If there exists something like that already, please point me to the correct location. Thanks!

**Comment by ritave** [comment]
Hey @Vid201 we're currently focused on getting Snaps to the main extension. No new features are being added before then.

I don't have any concrete dates for you but the best estimates we have is around middle of this year. Keep in mind these are not a promise in any capacity.

Our goal is for the v1 version to have stable API that we're going to support and not break long term. And to do that, we're doing all the breaking changes now, so that we don't need to support things we don't want to keep.

After getting to stable we're looking into different versioning schemes, one of which I like personally, is that the manifest version will dictate the backwards compatibility mode your snap will use.

**Reply by Vid201** [reply]
> Hey @Vid201 we're currently focused on getting Snaps to the main extension. No new features are being added before then.
> 
> I don't have any concrete dates for you but the best estimates we have is around middle of this year. Keep in mind these are not a promise in any capacity.
> 
> Our goal is for the v1 version to have stable API that we're going to support and not break long term. And to do that, we're doing all the breaking changes now, so that we don't need to support things we don't want to keep.
> 
> After getting to stable we're looking into different versioning schemes, one of which I like personally, is that the manifest version will dictate the backwards compatibility mode your snap will use.

Ok, thanks for explaining. Looking forward!

**Comment by Montoya** [comment]
We have a changelog in the Releases section of this repository: 
https://github.com/MetaMask/snaps-monorepo/releases

However, we cannot predict when a release of snaps-monorepo will be added to Flask. The release process is ad-hoc until we hit stable. We will try to make this process more predictable in stable.
